src/watch/index.ts

```diff
@@ -243,9 +244,13 @@ export class Task {
 					result
 				});
 			})
-			.catch((error: Error) => {
+			.catch((error: RollupError) => {
 				if (this.closed) return;
 
+				error.watchFiles.forEach(id => {
```

Comment:
May want to make this more restrictive so that it is only adding files in that initial error before any are added such as

```js
	if (this.watched.size === 0) {
	const watched = (this.watched = new Set());

	error.watchFiles.forEach(id => {
		watched.add(id);
		this.watchFile(id);
	});
}
```